src/__tests__/utilities/integration-test.utilities.js

```diff
@@ -0,0 +1,127 @@
+    validateExternalAttachmentEquality(original, modified)
+  ];
+
+  return conditions.every(test => test === true);
+};
+
+const validateEbXmlEquality = (original, modified) => {
+  const ebXMLs = {
+    original: parser.parse(JSON.parse(original).ebXML),
+    modified: parser.parse(JSON.parse(modified).ebXML)
+  };
+
+  // Check original vs modified ebXML
+  for (const key in ebXMLs) {
+    if (ebXMLs.hasOwnProperty(key)) {
+      // Delete the Message ID attribute within MessageData -> MessageId
+      delete ebXMLs[key]["Envelope"]["Header"]["MessageHeader"]["MessageData"]["MessageId"];
+
+      // Delete inner manifest Message ID (i.e. mid) references
+      if(Array.isArray(ebXMLs[key]["Envelope"]["Body"]["Manifest"]["Reference"])) {
+        ebXMLs[key]["Envelope"]["Body"]["Manifest"]["Reference"].forEach(reference => {
+          if (reference["@_href"].includes('mid')) {
+            delete reference["@_href"];
+          }
+        });
+      }
+    }
+  }
+
+  return isEqual(ebXMLs.original, ebXMLs.modified);
+}
+
+const validatePayloadEquality = (original, modified) => {
+  const payloads = {
+    original: parser.parse(JSON.parse(original).payload),
+    modified: parser.parse(JSON.parse(modified).payload)
+  };
+
+  for (const key in payloads) {
+    if (payloads.hasOwnProperty(key)) {
+      if (original.includes(INTERACTION_IDS.UK06) && modified.includes(INTERACTION_IDS.UK06)) {
+        // Remove Message ID references within the UK06 message
+        delete payloads[key][INTERACTION_IDS.UK06]['id']['@_root'];
+        delete payloads[key][INTERACTION_IDS.UK06]['ControlActEvent']['subject']['EhrExtract']['id']['@_root'];
+      }
+      else if (original.includes(INTERACTION_IDS.COPC) && modified.includes(INTERACTION_IDS.COPC)) {
+        // Remove Message ID references within the COPC message
+        delete payloads[key][INTERACTION_IDS.COPC]['id']['@_root'];
+        delete payloads[key][INTERACTION_IDS.COPC]['ControlActEvent']['subject']['PayloadInformation']['id']['@_root'];
+        delete payloads[key][INTERACTION_IDS.COPC]['ControlActEvent']['subject']['PayloadInformation']['value']['Gp2gpfragment']['message-id'];
+        delete payloads[key][INTERACTION_IDS.COPC]['ControlActEvent']['subject']['PayloadInformation']['pertinentInformation']['pertinentPayloadBody']['id']['@_root'];
+      }
+      else {
+        logError('Unrecognised Interaction ID');
+        return false;
+      }
+    }
+  }
+
+  return isEqual(payloads.original, payloads.modified);
+};
+
+const validateAttachmentEquality = (original, modified) => {
+  const attachments = {
+    original: JSON.parse(original).attachments,
+    modified: JSON.parse(modified).attachments
+  };
+
+  return isEqual(attachments.original, attachments.modified);
+};
+
+const validateExternalAttachmentEquality = (original, modified) => {
+  if (original.includes('external_attachments') && modified.includes('external_attachments')) {
```

Comment:
Could there be a case here where original includes external_attachments and modified doesn't and the equality resolves to true? I would switch this to an ||